google-cloud-storage/src/main/java/com/google/cloud/storage/PostPolicyV4.java

```diff
@@ -18,56 +18,123 @@
 
 import com.google.gson.JsonArray;
 import com.google.gson.JsonObject;
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.LinkedHashSet;
+import java.util.List;
 import java.util.Map;
 import java.util.Objects;
 import java.util.Set;
+import java.util.concurrent.TimeUnit;
 
 /**
- * Presigned V4 post policy.
+ * Presigned V4 post policy. Instances of {@code PostPolicyV4} include a URL and a map of fields
+ * that can be specified in an HTML form to submit a POST request to upload an object.
  *
- * @see <a href="https://cloud.google.com/storage/docs/xml-api/post-object">POST Object</a>
+ * <p>See <a href="https://cloud.google.com/storage/docs/xml-api/post-object">POST Object</a> for
+ * details of upload by using HTML forms.
+ *
+ * <p>See {@link Storage#generateSignedPostPolicyV4(BlobInfo, long, TimeUnit,
+ * PostPolicyV4.PostFieldsV4, PostPolicyV4.PostConditionsV4, Storage.PostPolicyV4Option...)} for
+ * example of usage.
  */
 public final class PostPolicyV4 {
   private final String url;
   private final Map<String, String> fields;
 
   private PostPolicyV4(String url, Map<String, String> fields) {
+    try {
+      new URL(url);
+    } catch (MalformedURLException e) {
+      throw new IllegalArgumentException(e);
```

Comment:
This needs a test

Reply:
The first portion of planned tests which cover the changed code has been added.

Reply:
@elharo all the planned tests are pushed